src/components/Connections/Connections.tsx

```diff
@@ -270,7 +270,11 @@ export const Connections = () => {
     setResetDeploymentId(connection.resetConnDeploymentId);
     setAnchorEl(event);
   };
-  const handleClose = () => {
+  const handleClose = (isEditMode?: string) => {
+    if(isEditMode !== "EDIT"){
+      setConnectionId("");
+      setResetDeploymentId("");
+    }
     setAnchorEl(null);
```

Comment:
why do we need the extra `isEditMode` param. Can't we just reset it whenever its called ? create or edit.

Reply:
-yes, but the problem is the form doesn't know when to reset.
- the `connectionId` cannot be solely used to check if its a Add new connection or edit connection.
- The reason is, The `CreateConnectionForm` depends on connectionId which is passed as a prop from parent `Connection` component. 
- So if click the (`3 dots)` and then **dont** click edit and close the optionsMenu by clicking anywhere on screen, the `connectionId` is still there. 
- so now when i click addNewConnection, it receives the connectionId and then populate the form with the data of that connectionId.